src/charm.py

```diff
@@ -1,36 +1,116 @@
 #!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
 
+""" A Juju Charm for Seldon Core Operator """
 import logging
+from base64 import b64encode
+from pathlib import Path
+from subprocess import check_call
 
+from charmed_kubeflow_chisme.kubernetes import KubernetesResourceHandler
+from charmed_kubeflow_chisme.lightkube.batch import delete_many
+from charms.observability_libs.v1.kubernetes_service_patch import KubernetesServicePatch
 from charms.prometheus_k8s.v0.prometheus_scrape import MetricsEndpointProvider
 from charms.grafana_k8s.v0.grafana_dashboard import GrafanaDashboardProvider
-from jinja2 import Environment, FileSystemLoader
+from lightkube.models.core_v1 import ServicePort
+from lightkube import ApiError
+from lightkube.generic_resource import load_in_cluster_generic_resources
 from ops.charm import CharmBase
+from ops.framework import StoredState
 from ops.main import main
-from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus
+from ops.model import ActiveStatus, MaintenanceStatus, WaitingStatus, BlockedStatus
+from ops.pebble import ChangeError, Layer
 
-from oci_image import OCIImageResource, OCIImageResourceError
+K8S_RESOURCE_FILES = [
+    "src/templates/auth_manifests.yaml.j2",
+    "src/templates/validate.yaml.j2",
+    "src/templates/webhook-service.yaml.j2",
+]
+CRD_RESOURCE_FILES = [
+    "src/templates/crd-v1.yaml.j2",
+]
+CONFIGMAP_RESOURCE_FILES = [
+    "src/templates/configmap.yaml.j2",
+]
+SSL_CONFIG_FILE = "src/templates/ssl.conf.j2"
 
-log = logging.getLogger()
 
+#
+# Helper for raising the exception
+#
+class CheckFailed(Exception):
+    """Raise this exception if one of the checks in main fails."""
+
+    def __init__(self, msg, status_type=None):
+        super().__init__()
+
+        self.msg = str(msg)
+        self.status_type = status_type
+        self.status = status_type(self.msg)
+
+
```

Comment:
A shared version of this is now available in chisme.  I think this would work:
```python
from charmed_kubeflow_chisme.exceptions import ErrorWithStatus as CheckFailed
```
(or just import as `ErrorWithStatus` and rename any usage of `CheckFailed` below)

Reply:
Done.